app/scripts/metamask-controller.js

```diff
@@ -2507,8 +2507,8 @@ export default class MetamaskController extends EventEmitter {
 
       // TODO: Remove once the preferences controller has been replaced with the core monorepo implementation
       this.controllerMessenger.publish(
-        `${this.preferencesController.name}:stateChange`,
-        [state, []],
+        'PreferencesController:stateChange',
+        state,
```

Comment:
The old code's state publish was coming through as an array, which did not destruct correctly.  I don't know if the proper fix is to publish an object, or deconstruct the first element of the array instead.

![image](https://github.com/MetaMask/metamask-extension/assets/3500406/fa1db3e3-e88a-468b-bf4a-312a421dc4cd)